sdk/core/System.ServiceModel.Rest/api/System.ServiceModel.Rest.net6.0.cs

```diff
@@ -155,39 +154,45 @@ public partial class HttpPipelineRequest : System.ServiceModel.Rest.Core.Pipelin
         protected virtual void AddHeader(string name, string value) { }
         protected virtual bool ContainsHeader(string name) { throw null; }
         public virtual void Dispose() { }
-        protected System.Collections.Generic.IEnumerable<string> GetHeaderNames() { throw null; }
         protected virtual bool RemoveHeader(string name) { throw null; }
         protected virtual void SetHeader(string name, string value) { }
         public override void SetHeaderValue(string name, string value) { }
         public virtual void SetMethod(System.Net.Http.HttpMethod method) { }
         public override void SetMethod(string method) { }
         public override string ToString() { throw null; }
         protected virtual bool TryGetHeader(string name, out string? value) { throw null; }
+        protected bool TryGetHeaderNames(out System.Collections.Generic.IEnumerable<string> headerNames) { throw null; }
         protected virtual bool TryGetHeaderValues(string name, out System.Collections.Generic.IEnumerable<string>? values) { throw null; }
         public virtual bool TryGetMethod(out System.Net.Http.HttpMethod method) { throw null; }
     }
     public partial class HttpPipelineResponse : System.ServiceModel.Rest.Core.PipelineResponse, System.IDisposable
     {
         public HttpPipelineResponse(System.Net.Http.HttpResponseMessage? httpResponse, System.IO.Stream? contentStream) { }
         public override System.IO.Stream? ContentStream { get { throw null; } set { } }
-        public override string ReasonPhrase { get { throw null; } }
         public override int Status { get { throw null; } }
         public override void Dispose() { }
         protected virtual void Dispose(bool disposing) { }
         public override System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> GetHeaders() { throw null; }
         public override bool TryGetHeaderValue(string name, out System.Collections.Generic.IEnumerable<string>? values) { throw null; }
         public override bool TryGetHeaderValue(string name, out string? value) { throw null; }
+        public override bool TryGetReasonPhrase(out string reasonPhrase) { throw null; }
+    }
+    public partial interface IPipelineEnumerator
```

Comment:
I made this an interface instead of an abstract class so we could make the enumerator types in the Azure.Core pipeline policies structs instead of classes to reduce allocations, but if this is not the right perf tradeoff, we can put it back.